### PR TITLE
Add extractors for cpu/mem metrics

### PR DESCRIPTION
**Pull request by pxaws**
**Description:** <Describe what has changed.>
To reduce the size of the PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3414, I create a separate PR to add the extractors for cpu/mem metrics. 

The extractors take the container stats generated by cadvisor as input and do some processing (e.g. rate calculation) to produce the metrics for Container Insights. 

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2307

**Testing:** <Describe what testing was performed and which tests were added.>
Unit tests.

### CODE REVIEW

**Review by mxiamxia — APPROVED** [review]
Thanks

**Review by Aneurysm9 — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]
Please make sure you think how this interacts with host metrics, and why we need the same metrics extracted in multiple places.

File: receiver/awscontainerinsightreceiver/internal/cadvisor/extractors/mem_extractor_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		cMetrics = extractor.GetValue(result[0], MockCPUMemInfo, containerType)
+	}
+
+	if extractor.HasValue(result2[0]) {
+		cMetrics = extractor.GetValue(result2[0], MockCPUMemInfo, containerType)
+	}
+
+	for _, cadvisorMetric := range cMetrics {
+		log.Printf("cadvisor Metrics received:\n %v \n", *cadvisorMetric)
+	}
```
Comment (bogdandrutu) [inline receiver/awscontainerinsightreceiver/internal/cadvisor/extractors/mem_extractor_test.go:44]: Why logging in unittests?

File: receiver/awscontainerinsightreceiver/internal/cadvisor/testutils/helpers.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		log.Printf("Fail to read file content: %s", err)
+	}
+
+	var result []*cinfo.ContainerInfo
+	containers := map[string]*cinfo.ContainerInfo{}
+	err = json.Unmarshal(info, &containers)
+
+	if err != nil {
+		log.Printf("Fail to parse json string: %s", err)
+	}
```
Comment (bogdandrutu) [inline receiver/awscontainerinsightreceiver/internal/cadvisor/testutils/helpers.go:38]: Consider to use assertions instead of logging in unittests.

**Review by bogdandrutu — CHANGES_REQUESTED** [review]
Please apply comments to all the unittests and testutils

File: receiver/awscontainerinsightreceiver/internal/cadvisor/extractors/mem_extractor_test.go (hunk 1 above)
Comment (pxaws) [reply receiver/awscontainerinsightreceiver/internal/cadvisor/extractors/mem_extractor_test.go:44]: I will remove it

**Comment by pxaws** [comment]
> Please make sure you think how this interacts with host metrics, and why we need the same metrics extracted in multiple places.

Host metrics is a different type of metrics in our case because it comes from different resources. We do the extraction at a different places and use it to enrich to cadvisor metrics. Host metrics and cadvisor metrics are not the same metrics.

**Comment by pxaws** [comment]
> Please apply comments to all the unittests and testutils

I have changed the tests accordingly

**Comment by bogdandrutu** [comment]
>> Host metrics is a different type of metrics in our case because it comes from different resources. We do the extraction at a different places and use it to decorate cadvisor metrics. Host metrics and cadvisor metrics are not the same metrics.

Very vague response

**Review by bogdandrutu — APPROVED** [review]

**Comment by pxaws** [comment]
> > > Host metrics is a different type of metrics in our case because it comes from different resources. We do the extraction at a different places and use it to decorate cadvisor metrics. Host metrics and cadvisor metrics are not the same metrics.
> 
> Very vague response

The cadvisor metrics are based on cgroup hierarchy and are stats about each containers (and its parent pods). To compute some percent metrics, we need to compare it with the host metrics.

**Pull request merged**